services/QuillLMS/client/app/bundles/Teacher/components/salesForm/index.tsx

```diff
@@ -26,28 +26,26 @@ export const SalesForm = ({ type }) => {
   const [schools, setSchools] = React.useState<any[]>([]);
   const [districts, setDistricts] = React.useState<any[]>([]);
   const [selectedSchool, setSelectedSchool] = React.useState<any>('');
+  const [schoolSearchQuery, setSchoolSearchQuery] = React.useState<any>('');
   const [schoolNotListed, setSchoolNotListed] = React.useState<boolean>(false);
   const [selectedDistrict, setSelectedDistrict] = React.useState<any>('');
+  const [districtSearchQuery, setDistrictSearchQuery] = React.useState<any>('');
   const [districtNotListed, setDistrictNotListed] = React.useState<boolean>(false);
   const [schoolOrDistrict, setSchoolOrDistrict] = React.useState<any>('');
   const [showSnackbar, setShowSnackbar] = React.useState(false)
 
   React.useEffect(() => {
-    if(!schools.length) {
-      getSchoolsAndDistricts('school').then((response) => {
-        if(response && response.options) {
-          setSchools(response.options);
-        }
-      });
-    }
-    if(!districts.length) {
-      getSchoolsAndDistricts('district').then((response) => {
-        if(response && response.options) {
-          setDistricts(response.options);
-        }
-      });
-    }
-  }, []);
+    getSchoolsAndDistricts('school', schoolSearchQuery).then((response) => {
+      if(response && response.options) {
+        setSchools(response.options);
+      }
+    });
+    getSchoolsAndDistricts('district', districtSearchQuery).then((response) => {
+      if(response && response.options) {
+        setDistricts(response.options);
+      }
+    });
+  }, [schoolSearchQuery, districtSearchQuery]);
```

Comment:
Does it make sense to split these two up and only do `getSchoolsAndDistricts('school',...` on `[schoolSearchQuery]` and same for districts? Because the value of one isn't dependent on the other, right? Otherwise I think we'll be unnecessarily hitting the server an extra time for every keystroke.

Reply:
yeah that's a good point, I'll separate them out